site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+    sectionName: store
+  - kind: HTTPRoute
+    name: my-example
+    namespace: prod-gw
+    sectionName: api
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /login
+    backendRefs:
+    - name: auth-service
+      port: 8080
+  - matches:
+    - path:
+        type: Pathprefix
+        value: /account
+    backendRefs:
+    - name: account-service
+      port: 8080
+status:
+  parents:
+  - parentRef:
+      kind: HTTPRoute
+      name: my-example
+      namespace: prod-gw
+      sectionName: store
+    controllerName: prod-gw
+    conditions:
+    - type: Attached
+      status: true
+      reason: Attached
+      message: Successfully attached to `store` section.      
+    grandparents:
+    - parentRef:
+          kind: Gateway
+          name: prod-web-gw
+        conditions:
+        - type: Accepted
+          status: true
+          reason: Accepted
+          Message: Route successfully attached to parent and gateway.
+  - parentRef:
+      kind: HTTPRoute
+      name: my-example
+      namespace: prod-gw
+      sectionName: api
+    controllerName: prod-gw
+    conditions:
+    - type: Attached
+      status: true
+      reason: Attached
+      message: Successfully attached to `api` section.
+    grandparents:
+    - parentRef:
+          kind: Gateway
+          name: prod-web-gw
+        conditions:
+        - type: Accepted
+          status: true
+          reason: Accepted
+          Message: Route successfully attached to parent and gateway.
+```
+
+This is pretty verbose, but the extra information will be very important once we
+begin having errors with our configuration.
+
+### Multiple Gateways, same GatewayClass, domain match error
+
+In this example, the HTTPRoute `http-parent` is able to attach to either Gateway by itself, but the Gateways share a GatewayClass (and `controllerName`).
+
+However, because both Gateways include different `hostname` fields, once the child is included, `gateway-a` will accept it, and `gateway-b` will not, because the hostnames don't match.
+
+This is a slightly contrived example, but serves to illustrate that we need a way for both `http-parent` and `http-child` to indicate that something has gone wrong.
+
+```yaml
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: gateway-a
+spec:
+  gatewayClassName: acme-lb
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 80
+    hostname: *.example.com
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: gateway-b
+spec:
+  gatewayClassName: acme-lb
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 80
+    hostname: subdomain.example.com
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-parent
+spec:
+  parentRefs:
+  - name: gateway-a
+    sectionName: http
+  - name: gateway-b
+    sectionName: http
+  rules:
+    - allowedRoutes:
+        kinds: 
+          - kind: HTTPRoute
```

Comment:
Why is this allowed?  I thought we could only route to the same type

Reply:
This is from an earlier version that allowed other Kinds, and hasn't been updated.